src/io/flutter/actions/RunFlutterAction.java

```diff
@@ -62,19 +62,18 @@ public void actionPerformed(AnActionEvent e) {
     final SdkRunConfig sdkRunConfig = (SdkRunConfig)configuration.clone();
     final SdkFields fields = sdkRunConfig.getFields();
     final String additionalArgs = fields.getAdditionalArgs();
-    final String runArg = "--" + myLaunchMode.getCliCommand();
-    String flavorArg = "";
+
+    String flavorArg = null;
     if (fields.getBuildFlavor() != null) {
-      flavorArg = " --flavor=" + fields.getBuildFlavor();
+      flavorArg = "--flavor=" + fields.getBuildFlavor();
     }
 
-    if (additionalArgs == null) {
-      fields.setAdditionalArgs(runArg + flavorArg);
-    }
-    else {
-      if (!additionalArgs.contains(runArg)) {
-        fields.setAdditionalArgs(additionalArgs + flavorArg + " " + runArg);
-      }
+    if (additionalArgs != null && flavorArg != null) {
```

Comment:
nit:  I think the intent is a little clearer if written like:
List<String> args = new ArrayList<>();
if (additionalArgs != null) {
  args.add(additionalArgs);
}
if (flavorArg != null) {
  args.add(flavorArg);
}
if (args.isNotEmpty()) {
  fields.setAdditionalArgs(joiner.on(" ").join(args));
}

Reply:
sounds good